File: speed.lua
local running = false
local timer

function show_current_speed()
    running = not running
    if running then
        timer = mp.add_periodic_timer(1, function()
            local currentspeed = mp.get_property("speed")
            mp.osd_message("Speed: "..("%.2f"):format(math.floor(currentspeed * 1000) / 1000), 1)
        end)
    else
        timer:stop()
    end
end

mp.add_key_binding("ctrl+s", "show_current_speed", show_current_speed)

File: speed.js
var running = false;
var timer;

function show_current_speed() {
    running = !running;
    if (running) {
        timer = setInterval(function() {
            var currentspeed = mp.get_property("speed");
            mp.osd_message("Speed: " + (Math.round(currentspeed * 100) / 100).toFixed(2), 1);
        }, 1000);
    } else {
        clearInterval(timer);
    }
}

mp.add_key_binding("ctrl+s", "show_current_speed", show_current_speed);
